### PR TITLE
Tech docs

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Somewhat lazily, I've tried to cover in one doc:
- a tech overview
- what it might make sense to extract or reuse elsewhere

### CODE REVIEW

File: docs/tech-overview.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+- enable debug features
+
+This system may change without notice. Flags are regularly added and removed.
+
+The current set of flags are documented in [the source](../src/flags.ts).
+
+Flags may be specified via the query string with repeated `flag` parameters,
+for example, http://localhost:3000/?flag=oneFlag&flag=anotherFlag
+
+By default, all flags are enabled for local development and branches builds.
```
Comment (microbit-robert) [inline docs/tech-overview.md:97]:
```suggestion
By default, all flags are enabled for local development and branch builds.
```

**Comment by microbit-robert** [comment]
We use different links for support, https://support.microbit.org/ in the readme and mailto:help@microbit.org in the tech-overview. Not sure if this is by design. 

Should we add a section on search and language support for search?

**Comment by microbit-matt-hillsdon** [comment]
> We use different links for support, https://support.microbit.org/ in the readme and [mailto:help@microbit.org](mailto:help@microbit.org) in the tech-overview. Not sure if this is by design.

Yeah, kinda, based on the readme likely having a more general audience. Not a very strong view though.

> Should we add a section on search and language support for search?

Makes sense, yep.

**Review by microbit-robert — APPROVED** [review]
LGTM.

**Pull request merged**